nearcache_map.go

```diff
@@ -21,23 +21,39 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/hazelcast/hazelcast-go-client/internal/logger"
+	inearcache "github.com/hazelcast/hazelcast-go-client/internal/nearcache"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto/codec"
 	"github.com/hazelcast/hazelcast-go-client/internal/serialization"
 	"github.com/hazelcast/hazelcast-go-client/nearcache"
+	"github.com/hazelcast/hazelcast-go-client/types"
+)
+
+const (
+	eventTypeInvalidation = 8
 )
 
 type nearCacheMap struct {
-	nc             *nearCache
+	nc             *inearcache.NearCache
 	toNearCacheKey func(key interface{}) (interface{}, error)
 	ss             *serialization.Service
+	lg             logger.LogAdaptor
 }
 
-func newNearCacheMap(nc *nearCache, ncc *nearcache.Config, ss *serialization.Service) (nearCacheMap, error) {
+func newNearCacheMap(ctx context.Context, nc *inearcache.NearCache, ncc *nearcache.Config, ss *serialization.Service, lg logger.LogAdaptor, name string, p *proxy, local bool) (nearCacheMap, error) {
 	ncm := nearCacheMap{
 		nc: nc,
 		ss: ss,
+		lg: lg,
 	}
 	if ncc.InvalidateOnChange() {
-		ncm.registerInvalidationListener()
+		lg.Debug(func() string {
+			return fmt.Sprintf("registering invalidation listener: name: %s, local: %t", name, local)
+		})
+		if err := ncm.registerInvalidationListener(ctx, name, p, local); err != nil {
+			return nearCacheMap{}, fmt.Errorf("hazelcast.newNearCacheMap: preloading near cache: %w", err)
```

Comment:
Is this error message correct?

Reply:
Fixed at: https://github.com/hazelcast/hazelcast-go-client/pull/807/commits/ebcf2b3b7fe69e1af93821996e0b1ea1d524c69d